packages/number-input/src/NumberInput/NumberInput.spec.tsx

```diff
@@ -243,6 +230,39 @@ describe('packages/number-input', () => {
       expect((numberInput as HTMLInputElement).value).toBe('-1');
     });
 
+    describe('onBlur', () => {
+      test('callback triggers when focus leaves number input', () => {
+        const { numberInput } = renderNumberInput({
+          label,
+          ...defaultProps,
+        });
+
+        userEvent.tab(); // focus
+        expect(numberInput).toHaveFocus();
+        userEvent.tab(); // blur
+
+        expect(defaultProps.onBlur).toHaveBeenCalledTimes(1);
+      });
+
+      test('callback triggers when focus leaves arrow buttons', () => {
+        const { container, numberInput } = renderNumberInput({
+          label,
+          ...defaultProps,
+        });
+
+        const upArrow = container.querySelector(
```

Comment:
Maybe we should add testids to the arrows so this its easier for consumers to test?

Reply:
great, added those test ids and updated the relevant specs